dali/kernels/tensor_test.cc

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "dali/kernels/shape.h"
+#include "dali/kernels/tensor_view.h"
+
+namespace tensor {
+
+TEST(TensorShapeTest, StaticShapeConstructor) {
+  TensorShape<0> zero_tensor;
```

Comment:
Do we even allow this? It hardly makes sense and would break the multi-argument constructor `(Index i0, Indices...)`.

Reply:
It works unless someone is trying to is that constructor which won't happen for ndim=0. It may be useful in some templates.